tests/security_test.go

```diff
@@ -0,0 +1,146 @@
+package tests
+
+import (
+	"github.com/mesosphere/kudo-spark-operator/tests/utils"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+type serviceAccountTestCase struct {
+	name               string
+	namespace          string
+	params             map[string]string
+	expectedOperatorSA string
+	prepareOperatorSA  bool
+	expectedDriverSA   string
+	prepareDriverSA    bool
+}
+
+func TestServiceAccounts(t *testing.T) {
+	testCases := []serviceAccountTestCase{
+		{
+			name:      "DefaultConfiguration",
+			namespace: "sa-test-default",
+			params: map[string]string{
+				"operatorServiceAccountName": "spark-operator-service-account",
+				"sparkServiceAccountName":    "spark-service-account",
+			},
+			expectedOperatorSA: utils.DefaultInstanceName + "-spark-operator-service-account",
+			expectedDriverSA:   utils.DefaultInstanceName + "-spark-service-account",
+		},
+		{
+			name:      "ProvidedOperatorServiceAccount",
+			namespace: "sa-test-operator",
+			params: map[string]string{
+				"operatorServiceAccountName":   "custom-operator-sa",
+				"createOperatorServiceAccount": "false",
+				"sparkServiceAccountName":      "spark-service-account",
+			},
+			prepareOperatorSA:  true,
+			expectedOperatorSA: "custom-operator-sa",
+			expectedDriverSA:   utils.DefaultInstanceName + "-spark-service-account",
+		},
+		{
+			name:      "ProvidedSparkServiceAccount",
+			namespace: "sa-test-spark",
+			params: map[string]string{
+				"operatorServiceAccountName": "spark-operator-service-account",
+				"createSparkServiceAccount":  "false",
+				"sparkServiceAccountName":    "custom-spark-sa",
+			},
+			prepareDriverSA:    true,
+			expectedOperatorSA: utils.DefaultInstanceName + "-spark-operator-service-account",
+			expectedDriverSA:   "custom-spark-sa",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			err := runServiceAccountTestCase(tc)
+			if err != nil {
+				t.Errorf("Test case: %v\nfailed with error: %s", tc, err)
+			}
+		})
+	}
+}
+
+func runServiceAccountTestCase(tc serviceAccountTestCase) error {
```

Comment:
The method still looks too long on my taste. But overall the PR's code looks really great now! Many thanks for the refactorings, @rpalaznik!